code/modules/admin/verbs/deadsay.dm

```diff
@@ -47,3 +39,47 @@
 		return
 
 	dsay(msg)
+
+///Dsay but larger
+/client/proc/dooc(msg as text)
+	set category = "Admin.Events"
+	set name = "Dooc"
+	set hidden = TRUE
+	if(!admin_holder || !(admin_holder.rights & R_MOD))
+		to_chat(src, SPAN_DANGER("Only administrators may use this command."))
+		return
+	if(!mob)
+		return
+	if(prefs.muted & MUTE_DEADCHAT)
+		to_chat(src, SPAN_DANGER("You cannot send DOOC messages (muted)."))
+		return
+
+	if(!(prefs.toggles_chat & CHAT_DEAD))
+		to_chat(src, SPAN_DANGER("You have deadchat muted."))
+		return
+
+	if (handle_spam_prevention(msg, MUTE_DEADCHAT))
+		return
+
+	var/stafftype = null
+
+	stafftype = "[admin_holder.rank]"
+
+	msg = strip_html(msg)
+	log_admin("DEAD: [key_name(src)] : [msg]")
+
+	if (!msg)
+		return
+
+	var/rendered = "<span class='dooc'><span class='prefix'>DEAD:</span> <span class='name'>[stafftype] ([src.key])</span> says, <span class='message'>\"[msg]\"</span></span>"
+
+	show_to_deadsay(rendered)
+
+/client/proc/show_to_deadsay(message)
+	for (var/mob/mob in GLOB.observer_list)
```

Comment:
Did you mean to change this to observer? If so the admin check below on 81 I think is moot.